content/source/docs/extend/debugging.html.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: "extend"
+page_title: "Extending Terraform: Debugging Providers"
+sidebar_current: "docs-extend-debugging-providers"
+description: |-
+  How to debug providers.
+---
+
+# Debugging Providers
+
+Terraform's providers are complicated programs in their own right, and
+sometimes they don't end up working the way you expect them to. In those
+situations, it's not always obvious how to get information about what's
+actually happening inside a provider or why things are behaving the way they
+do.
```

Comment:
```suggestion
```

I don't think we need this motivating paragraph. If you've reached this document then you probably know what debugging is, and the next paragraph will tell you whether this document is meant for you.